website/project/views/drafts.py

```diff
@@ -0,0 +1,217 @@
+    return {
+        'status': 'initiated',
+        'urls': {
+            'registrations': node.web_url_for('node_registrations')
+        }
+    }, http.ACCEPTED
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registration(auth, node, draft, *args, **kwargs):
+    return serialize_draft_registration(draft, auth), http.OK
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registrations(auth, node, *args, **kwargs):
+
+    count = request.args.get('count', 100)
+    drafts = node.draft_registrations.find(
+        Q('registered_node', 'eq', None)
+    )[:count]
+    return {
+        'drafts': [serialize_draft_registration(d, auth) for d in drafts]
+    }, http.OK
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def create_draft_registration(auth, node, *args, **kwargs):
+
+    data = request.get_json()
+
+    schema_name = data.get('schema_name')
+    if not schema_name:
+        raise HTTPError(http.BAD_REQUEST)
+
+    schema_version = data.get('schema_version', 1)
+    # TODO(hrybacki): Move to framework.utils.rapply once @sam's PR#4027 is merged.
+    #from api.base.serializers import _rapply
+    # schema_data = _rapply(data.get('schema_data', {}), sanitize.strip_html)
+    schema_data = data.get('schema_data', {})
+
+    meta_schema = get_schema_or_fail(
+        Q('name', 'eq', schema_name) &
+        Q('schema_version', 'eq', schema_version)
+    )
+    draft = node.create_draft_registration(auth.user, meta_schema, schema_data, save=True)
+    return serialize_draft_registration(draft, auth), http.CREATED
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def new_draft_registration(auth, node, *args, **kwargs):
+
+    data = request.values
+
+    schema_name = data.get('schema_name')
+    if not schema_name:
+        raise HTTPError(
+            http.BAD_REQUEST,
+            data={
+                'message_short': 'Must specify a schema_name',
+                'message_long': 'Please specify a schema_name'
+            }
+        )
+
+    schema_version = data.get('schema_version', 1)
+
+    meta_schema = get_schema_or_fail(
+        Q('name', 'eq', schema_name) &
+        Q('schema_version', 'eq', int(schema_version))
+    )
+    draft = node.create_draft_registration(auth.user, meta_schema, {}, save=True)
+    return redirect(node.web_url_for('edit_draft_registration_page', draft_id=draft._id))
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def edit_draft_registration_page(auth, node, draft, **kwargs):
+    ret = project_utils.serialize_node(node, auth, primary=True)
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def update_draft_registration(auth, node, draft, *args, **kwargs):
+    data = request.get_json()
+
+    # TODO(hrybacki): Move to framework.utils.rapply once @sam's PR#4027 is merged.
```

Comment:
TODO can be removed.

It appears that sanitization has been removed entirely from this view, which is not optimal. We should be sanitizing fields that are saved in the database, particularly the fields of `schema_data`.